File: app/src/main/java/dev/ragnarok/fenrir/api/model/VKApiWallReply.java
package dev.ragnarok.fenrir.api.model;

import androidx.annotation.NonNull;

import com.google.gson.annotations.SerializedName;

public class VKApiWallReply implements VKApiAttachment {

    @SerializedName("id")
    public int id;

    @SerializedName("from_id")
    public int from_id;

    @SerializedName("post_id")
    public int post_id;

    @SerializedName("owner_id")
    public int owner_id;

    @SerializedName("text")
    public String text;

    @SerializedName("attachments")
    public VKApiAttachments attachments;

    public int getAttachmentsCount() {
        return attachments == null ? 0 : attachments.size();
    }

    public boolean hasAttachments() {
        return getAttachmentsCount() > 0;
    }

    @NonNull
    @Override
    public String getType() {
        return TYPE_WALL_REPLY;
    }
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/api/model/VKApiWallReply.kt
package dev.ragnarok.fenrir.api.model

import dev.ragnarok.fenrir.orZero
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class VKApiWallReply : VKApiAttachment {
    @SerialName("id")
    var id = 0

    @SerialName("from_id")
    var from_id = 0

    @SerialName("post_id")
    var post_id = 0

    @SerialName("owner_id")
    var owner_id = 0

    @SerialName("text")
    var text: String? = null

    @SerialName("attachments")
    var attachments: VKApiAttachments? = null
    val attachmentsCount: Int
        get() = attachments?.size().orZero()

    fun hasAttachments(): Boolean {
        return attachmentsCount > 0
    }

    override fun getType(): String {
        return VKApiAttachment.TYPE_WALL_REPLY
    }
}